src/server/string_family.cc

```diff
@@ -1294,22 +1361,39 @@ void StringFamily::DecrBy(CmdArgList args, const CommandContext& cmnd_cntx) {
   return IncrByGeneric(key, -val, cmnd_cntx.tx, cmnd_cntx.rb);
 }
 
+std::optional<DbSlice::ExpireParams> ParseExpireParams(CmdArgList& args) {
+  std::optional<DbSlice::ExpireParams> expire_params = std::nullopt;
+  if (args.size() > 2) {
+    if (args[args.size() - 2] == "EXAT") {
+      int64_t expiry;
+      absl::SimpleAtoi(args[args.size() - 1], &expiry);
+      // Memcache expire time is absolute seconds
+      expire_params =
+          DbSlice::ExpireParams{.value = expiry, .absolute = true, .persist = expiry == 0};
+      args = args.subspan(0, args.size() - 2);
```

Comment:
This is incorrect because the callback works on shard args, not this list. Also the expiry we add `EXAT ...` gets added to shard args. In `MGet` we use shard args to construct the result, which causes a mismatch.

It might be simpler to store the expiry in the connection context like the memcache flag is stored, instead of passing it as extra args to mget command.